torch_xla/experimental/pjrt.py

```diff
@@ -224,3 +226,42 @@ def broadcast_master_param(model: nn.Module) -> None:
     parameters_and_buffers.append(p.data)
   xm.all_reduce(xm.REDUCE_SUM, parameters_and_buffers)
   xm.mark_step()
+
+
+def rendezvous(tag: str, payload: bytes,
+               ordinals: Optional[List[int]]) -> List[bytes]:
+  """Share `payload` with all replicas in `ordinals`.
+
+  All of PjRt is experimental right now, but consider `rendezvous` to be _very_
+  experimental. Only tested on TPU v4.
+
+  `tag` is ignored except for logging.
+
+  Args:
+    tag: Name of this rendezvous operation.
+    payload: Payload to share with other replicas.
+    ordinals: List of replicas participating in rendezvous.
+  Returns:
+    List of bytes from other replicas.
+  """
+  if not dist.is_initialized():
+    logging.warning(
+        'Default process group not initialized. Creating XLA process group...')
+    mesh_master = xu.getenv_as("XRT_MESH_SERVICE_ADDRESS", str)
+
+    import torch_xla.distributed.xla_backend
+    dist.init_process_group(
```

Comment:
would this conflict with ddp in any way?

Reply:
It could cause an issue if someone were to run rendezvous before they tried to initialize the main process group with another configuration. Otherwise, I added a check here to make sure we don't initialize the process group twice.

Reply:
hmm on a second thought, this backend is gloo, while on dpp we use the `xla_backend`. In a ddp setup I would expect two of them not conflict with each other?

Reply:
@alanwaketan to confirm my statement above.

Reply:
I followed the example in the PyTorch docs that initialized a `nccl` default group and a `gloo` subgroup to use `barrier`, so I assume this is safe: https://pytorch.org/docs/stable/distributed.html#debugging-torch-distributed-applications.

The default group that this function creates also uses the `xla` backend, which does seem to work fine in the tests.